### DISCUSSION TITLE
New to Plugin: Any working examples to understand the logic behind panel creation?

### DISCUSSION THREAD

**Discussion by HorselessName** [gapitio/gapit-htmlgraphics-panel#123] [Q&A]
Hello there!

I'm finding quite difficult to understand the logic behind the panel. I know how to make simple stuff like single panels, etc. but when trying to work with tables i am not figuring it ou how to do it properly.

PS: I am reading the documentation but still didnt understand.

Below a example of what i'm trying to achieve:

![image](https://user-images.githubusercontent.com/20173635/191831327-3523f840-2e28-4b42-8f22-6fe146f3db8d.png)

This is the table i receive from the query, every column being a specific service (That will become #1, #2 square as per image above):

![image](https://user-images.githubusercontent.com/20173635/191831548-b9303355-d4cf-4201-bb3c-19695f0d1fd9.png)

I already achieved the partial result using "Stats" plugin but sadly it doesnt allow me to nest/group panels and design accordingly to the image above, so i opted by going with HTML Graphics plugin...

But i really am stuck at Step 1 on how to do this. I know that it's possible, sure, but can someone give me some light?
Or examples even, so i can understand the logic behind it.

Thanks in advance!

**Comment by ZuperZee** [comment]
The plugin is a bit rough in the beginning as it's a lot to learn.
The easiest way I find doing stuff is to `console.log(data)` and find what I want in there.

To get the column name (the names on top in the table), you need to go to `data.series[x].fields[x].name`.

A simple example of how to display column names:
`Queries`
![image](https://user-images.githubusercontent.com/39485579/192007521-2a5cca0f-5912-4440-b523-b2e7afdeba04.png)
`Table view`
![image](https://user-images.githubusercontent.com/39485579/192007349-bf1efe1a-5126-4ca2-9c14-4822cb324e01.png)
`Normal view`
![image](https://user-images.githubusercontent.com/39485579/192006966-ecf8b573-8687-42b5-ae8a-a957b8a3d737.png)

`Panel options`
```json
{
    "calcsMutation": "standard",
    "reduceOptions": {
        "calcs": [
            "lastNotNull",
            "last",
            "firstNotNull",
            "first",
            "min",
            "max",
            "mean",
            "sum",
            "count",
            "range",
            "delta",
            "step",
            "diff",
            "logmin",
            "allIsZero",
            "allIsNull",
            "diffperc"
        ]
    },
    "add100Percentage": true,
    "centerAlignContent": true,
    "overflow": "visible",
    "useGrafanaScrollbar": true,
    "SVGBaseFix": true,
    "codeData": "{\n  \"text\": \"Random text\"\n}",
    "rootCSS": "",
    "css": "* {\n  font-family: Open Sans;\n}\n\n#square-container {\n  display: flex;\n}\n\n.square {\n  padding: 1rem;\n  margin-right: 0.5rem;\n  border-radius: 6px;\n  background-color: lightgreen;\n}\n\n.theme-dark .square {\n  background-color: darkgreen;\n}\n",
    "html": "<div>\n  <h2>\n    Columns\n  </h2>\n  <div id=\"square-container\">\n  </div>\n</div>\n",
    "renderOnMount": true,
    "onRender": "const squareContainer = htmlNode.querySelector(\"#square-container\")\r\nsquareContainer.innerHTML = \"\"; // Remove previous squares\r\n\r\n\r\nfunction createSquare(text) {\r\n  const square = document.createElement(\"div\");\r\n  square.classList.add(\"square\");\r\n  square.textContent = text;\r\n  squareContainer.appendChild(square);\r\n}\r\n\r\nconst [firstSeries] = data.series;\r\n\r\nif (firstSeries) {\r\n  for (const field of firstSeries.fields) {\r\n    if (field.name !== \"Time\") {\r\n      createSquare(field.name);\r\n    }\r\n  }\r\n}\r\nconsole.log(data)",
    "dynamicHtmlGraphics": false,
    "dynamicData": false,
    "dynamicFieldDisplayValues": false,
    "dynamicProps": false,
    "panelupdateOnMount": true,
    "onInitOnResize": false,
    "onInit": "if (htmlGraphics.theme2.isDark) {\r\n  htmlNode.lastElementChild.classList.add(\"theme-dark\")\r\n}\r\n"
}
```
